src/libostree/ostree-kernel-args.h

```diff
@@ -19,41 +19,90 @@
 
 #pragma once
 
-#include "libglnx.h"
+#include <gio/gio.h>
+#include <glib.h>
+#include <glib-object.h>
+#include "ostree-types.h"
 
 G_BEGIN_DECLS
 
 typedef struct _OstreeKernelArgs OstreeKernelArgs;
-void _ostree_kernel_args_free (OstreeKernelArgs *kargs);
-G_DEFINE_AUTOPTR_CLEANUP_FUNC(OstreeKernelArgs, _ostree_kernel_args_free);
-
-OstreeKernelArgs *_ostree_kernel_args_new (void);
-void _ostree_kernel_args_replace_take (OstreeKernelArgs  *kargs,
-                                       char              *key);
-void _ostree_kernel_args_replace (OstreeKernelArgs  *kargs,
-                                  const char        *key);
-void _ostree_kernel_args_replace_argv (OstreeKernelArgs  *kargs,
+
+_OSTREE_PUBLIC
+void ostree_kernel_args_free (OstreeKernelArgs *kargs);
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(OstreeKernelArgs, ostree_kernel_args_free)
+
+_OSTREE_PUBLIC
+GHashTable* ostree_kernel_arg_get_kargs_table (OstreeKernelArgs *kargs);
+
+_OSTREE_PUBLIC
+GPtrArray* ostree_kernel_arg_get_key_array (OstreeKernelArgs *kargs);
```

Comment:
So, I'm thinking we shouldn't expose these two functions here because of #1859. It's possible that we may change the data structures we use underneath.

Reply:
Yea I've noticed the data structure as mentioned in #1859, will be making them private and thanks for reviewing !